### PR TITLE
[NUI] Add AppBar class

### PR DESCRIPTION
**Pull request by jaehyun0cho**
AppBar shows title text and provides navigation and action functions on
Page.

### API Changes ###

 - ACR:

### CODE REVIEW

File: src/Tizen.NUI.Components/Controls/Navigation/AppBar.cs [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+using System.Windows.Input;
+
+namespace Tizen.NUI.Components
+{
+    /// <summary>
+    /// The AppBar class is a class which shows title text and provides navigation
+    /// and action functions on Page.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class AppBar : Control
```
Comment (hyunjushin) [inline src/Tizen.NUI.Components/Controls/Navigation/AppBar.cs:31]: Can you check similar class name from other platform?

Comment (jaehyun0cho) [reply src/Tizen.NUI.Components/Controls/Navigation/AppBar.cs:31]: Basically, I refers material design's app bar. (https://material.io/components/app-bars-top)

Class names from other platforms are as follows.
- Flutter : AppBar (https://api.flutter.dev/flutter/material/AppBar-class.html)
- Android : Toolbar (https://developer.android.com/reference/androidx/appcompat/widget/Toolbar)
- IOS : UINavigationBar (https://developer.apple.com/documentation/uikit/uinavigationbar)

**Review by AdunFang — APPROVED** [review]
It is OK.

File: src/Tizen.NUI.Components/Controls/Navigation/AppBar.cs [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+                TitleContent = titleContent;
+            }
+
+            if (actionContent != null)
+            {
+                ActionContent = actionContent;
+            }
+        }
+
+		/// <summary>
```
Comment (dongsug-song) [inline src/Tizen.NUI.Components/Controls/Navigation/AppBar.cs:108]: just small thing. indentation ^^;

could you check the sites below for easy way of editing code?

- Visual Studio in Windows : https://stackoverflow.com/questions/1435691/visual-studio-format-entire-file

- VS Code in Ubuntu : https://stackoverflow.com/questions/29973357/how-do-you-format-code-in-visual-studio-code-vscode

**Comment by dongsug-song** [comment]
@hyunjushin 
Hello, 
could you review again? (I think it's good to go.)
you have commented here, so I am just asking. 
thank you.

**Review by hyunjushin — APPROVED** [review]

**Comment by rabbitfor** [comment]
Hi @jaehyun0cho 
Thank you for your work.
Since this feature is new so it has a lot of APIs to review.
I'd be grateful if you provided picture that shows AppBar's structure.
(I ran the sample, it worked great! but it shows only one case.)

For example, 
스크린샷 2020-12-24 오후 6 10 01

**Comment by dongsug-song** [comment]
@jaehyun0cho 
Hello, could you check @rabbitfor 's review and comment above?
after @rabbitfor accepts, I will merge this PR. 
thank you.

File: src/Tizen.NUI.Components/Controls/Navigation/AppBar.cs (hunk 2 above)
Comment (jaehyun0cho) [reply src/Tizen.NUI.Components/Controls/Navigation/AppBar.cs]: Thank you for the tip! I have fixed the indentation.

**Comment by jaehyun0cho** [comment]
@rabbitfor @dongsug-song 

The components of AppBar are as follows.

image

- AppBar is located on the top of Page.
  - Page provides the property AppBar.
- NavigationContent is placed on the fat left.
  - If AutoNavigationContent is true, then default NavigationContent (back button) is displayed and it pops the page when it is clicked.
  - Custom NavigationContent can be set by user. (i.e. it is not necessary to be back button.)
- TitleContent is placed on the right of NavigationContent.
  - If AppBar(..., string title, ...) is used, TitleContent is filled with TextLabel with the given string.
  - Custom TitleContent can be set by user. (i.e. it is not necessary to be text.)
- ActionContent is placed on the far right.
  - If AppBar(..., params View[] actionContents) is used, the list of contents are set into a Control and the Control is set to ActionContent.
  - Custom ActionContent can be set by user.
- Overflow menu is not supported. (it can be considered if it is required.)
  - For now, user needs to add Overflow menu into ActionContent if it is required.

**Review by rabbitfor — COMMENTED** [review]
@jaehyun0cho 
Thank you for the information, it really helped to understand functionalities at a glance.
I think the APIs are OK.
If you don't mind, could you reordering APIs in the class?

Within a class,
- Fields
- Constructors
- Finalizers (Destructors)
- Delegates
- Events
- Enums
- Interfaces
- Properties
- Indexers
- Methods
- Structs
-Classes


Within each of these groups order by access
- public
- internal
- protected internal
- protected
- private protected
- private

**Review by AdunFang — APPROVED** [review]
It is OK.

**Comment by jaehyun0cho** [comment]
@rabbitfor 
Of course, I don't mind. I am sorry about this API orders. I have reordered the APIs.
About Page.cs, I will make another PR to reorder the events.
Thank you.

**Review by rabbitfor — APPROVED** [review]
LGTM

**Pull request merged**